### DISCUSSION TITLE
VIte error - getaddrinfo ENOTFOUND localhost

### DISCUSSION THREAD

**Discussion by saomartinho** [storybookjs/storybook#26887] [Help]
### Summary

Hello, 

 I'm migrating my storybook to Vite, and sometimes this error appear: 

```
info => Starting preview..
=> Failed to build the preview
Error: getaddrinfo ENOTFOUND localhost
    at node_internal_captureLargerStackTrace (node:internal/errors:563:5)
    at nodeinternal (node:internal/errors:782:10)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:internal/dns/promises:82:17)
```

Do you know what this can be?

**Comment by ZacharyDraper** [comment]
I was having this issue also, but I resolved it. While this may not be your issue, it's worth looking at.

TL/DR: If you are on a VPN, try disconnecting it and restart Storybook.

Storybook relies on a connection to localhost. You should have an entry for localhost in your hosts file (e.g. /etc/hosts). You can quickly check this by running `ping localhost` in your terminal. You should get back a response. In my case, my VPN client overwrites the hosts file and it does not put in an entry for localhost, so when I am connected to my VPN, a ping to localhost is unsuccessful which in turn means connecting to Storybook is unsuccessful. Interestingly, my connection to other localhost resources, like Docker, works fine either way.

**Comment by SanjeevAshoka** [comment]
Add below code to your vite.config file 
  **server: {
    port: 4200,       
    host: '127.0.0.1'  
  }**
 Your  Final Vite.config file should looks like this
 export default defineConfig({
  plugins: [react()],
.....your other configs...,
  server: {
    port: 4200,        // Set the port you want/ or available port
    host: '127.0.0.1'  // Set the host you want
  }
})
  This works for me.
